src/_blog/rust-libp2p-browser-to-server.md

```diff
@@ -0,0 +1,61 @@
+---
+tags:
+  - rust-libp2p
+  - rust
+  - libp2p
+  - browser
+  - webrtc
+  - webassembly
+  - wasm
+title: Rust-libp2p in the Browser with WebRTC!
+description: News about how rust-libp2p now connects with the network via WebRTC
+date: 2023-11-07
+permalink: ''
+translationKey: ''
+header_image: /rust-libp2p.jpeg
+author: DougAnderson444
+---
+
+# Rust-libp2p WebRTC now available in the Browser
+
+WebAssembly WebRTC is now available for use as a browser transport with rust-libp2p! Before now, WebRTC was only available on the server in `rust-libp2p`, but after months of coding and reviews, we are pleased to annouce it is available on both the server and in the browser.
+
+## Why WebRTC for libp2p?
+
+### Data Channels
+
+WebRTC's main purpose in peer-to-peer communications is for connecting media (audio and video). However, there is also a powerful [data channel](https://webrtc.org/getting-started/data-channels) API available which allows users to exchange data streams instead of audio/video. This is what libp2p uses at the transport layer. By tapping into this data channel API, we have been able to build libp2p Transports over WebRTC.
+
+### Server WebRTC vs Browser WebRTC
+
+Using WebRTC on the server involves using a Rust crate (currently [webrtc-rs](https://github.com/webrtc-rs/webrtc)), but in the browser it is available to Rust through browser bindings from a crate called [`web-sys`](https://docs.rs/web-sys/latest/web_sys/). By tying the libp2p code to these WebRTC bindings, we have been able to connect in the browser over WebRTC in pure Rust compiled to WebAssembly!
+
+### Benefits of WebRTC over websockets and webtransport
+
+One of the main benefits of using WebRTC on the server is that the it does not need TLS certificates from a DNS domain -- you can run a home server over WebRTC without a dotcom! Instead, with `rust-libp2p` we can use self-signed certificates to provide the secure connections. Websockets however would require TLS certificates.
```

Comment:
Note that authorities provide TLS certificates for IPs, not just domains. Overall explanation seems clunky. I like the "self-signed" wording. Maybe you can copy parts of https://blog.libp2p.io/libp2p-webrtc-browser-to-server/#self-signed-certificate?

Reply:
Thanks for these, I will re-word it to make it less clunky. I am trying to convey that there should be fewer steps and resources to set up a WebRTC node, say, at home, compared to a Websocket secure node.